### ISSUE TITLE
FR: Add List Indentation Standardization Feature (Tab-based)

### ISSUE THREAD

**Issue by justanotherjurastudent** [platers/obsidian-linter#1327]
**Title:** Add List Indentation Standardization Feature (Tab-based)

## Is Your Feature Request Related to a Problem? Please Describe.

I'm always frustrated when copying content from external sources (especially AI-generated responses like ChatGPT, Claude, or other websites) because the list items have inconsistent or incorrect indentation. AI responses typically use 2-4 spaces per indentation level instead of tabs, and copied content often has minimal indentation that doesn't follow Obsidian's standard tab-based convention. This results in poorly formatted lists that lose their visual hierarchy and are hard to read, requiring manual reformatting of every imported list.

## Describe the Solution You'd Like

I'd like a linting rule that automatically reformats list indentation to use Obsidian's standard tab-based approach. The feature should detect both ordered and unordered list items, analyze the logical hierarchy based on existing indentation patterns, and replace any existing indentation (spaces, mixed tabs/spaces) with standardized tab characters where each level uses exactly one tab character per indentation level.

**Please include an example where applicable:**

``` markdown
- Main item
  - Sub item (2 spaces)
    - Deep sub item (4 spaces)
      - Very deep item (6 spaces)
- Another main item
 - Inconsistent sub item (1 space)
1. Numbered list
   2. Sub numbered (3 spaces)
```

``` markdown
- Main item
	- Sub item (1 tab)
		- Deep sub item (2 tabs)
			- Very deep item (3 tabs)
- Another main item
	- Consistent sub item (1 tab)
1. Numbered list
	2. Sub numbered (1 tab)
```

## Describe Alternatives You've Considered

- **Manual reformatting:** Time-consuming and error-prone, especially with long documents
- **Obsidian's built-in tab functionality:** Only works when creating new lists, not when importing existing content
- **Find and replace:** Too simplistic and doesn't handle the logical hierarchy correctly
- **External tools:** Would require switching between applications and break the workflow
- **Custom scripts:** Would need to be maintained separately and lack integration with the linter

## Additional Context

This feature would be particularly valuable for users who frequently import content from AI tools or external sources. When using AI assistants to generate content, the responses consistently use space-based indentation instead of Obsidian's tab-based standard. The feature should be robust enough to handle mixed content (lists interspersed with regular text), various list markers (`-`, `*`, `+`, numbered lists), and edge cases like malformed indentation while preserving non-list content unchanged.

The implementation would complement the existing linting rules by ensuring that imported content follows Obsidian's native formatting conventions, making the linter a more comprehensive solution for markdown standardization.

**Reference Implementation:**

I've already created a working implementation for this feature as a standalone plugin. Here's the core logic that could be adapted for the linter:

```typescript
reformatListIndentation(editor: Editor) {
    const doc = editor.getDoc();
    const content = doc.getValue();
    const lines = content.split('\n');
    const newLines: string[] = [];

    // Stack to track indentation levels of current list hierarchy
    const indentLevelMarkers: number[] = [];
    let currentLogicalLevel = 0;

    for (const line of lines) {
        const trimmedLine = line.trimStart();
        const leadingWhitespace = line.length - trimmedLine.length;

        // Check if this is a list item
        const isListItem = (trimmedLine.startsWith('- ') ||
                            trimmedLine.startsWith('* ') ||
                            trimmedLine.startsWith('+ ') ||
                            /^\d+\.\s/.test(trimmedLine));

        if (isListItem) {
            if (indentLevelMarkers.length === 0) {
                // First list item in sequence
                indentLevelMarkers.push(leadingWhitespace);
                currentLogicalLevel = 0;
            } else {
                const lastIndentInStack = indentLevelMarkers[indentLevelMarkers.length - 1];

                if (leadingWhitespace > lastIndentInStack) {
                    // New deeper level detected
                    indentLevelMarkers.push(leadingWhitespace);
                    currentLogicalLevel = indentLevelMarkers.length - 1;
                } else if (leadingWhitespace  0 && leadingWhitespace  0 && leadingWhitespace === indentLevelMarkers[indentLevelMarkers.length - 1]) {
                        currentLogicalLevel = indentLevelMarkers.length - 1;
                    } else if (indentLevelMarkers.length > 0 && leadingWhitespace > indentLevelMarkers[indentLevelMarkers.length - 1]) {
                        indentLevelMarkers.push(leadingWhitespace);
                        currentLogicalLevel = indentLevelMarkers.length - 1;
                    } else {
                        // Reset and start new list
                        indentLevelMarkers.length = 0;
                        indentLevelMarkers.push(leadingWhitespace);
                        currentLogicalLevel = 0;
                    }
                } else {
                    // Same level as previous item
                    currentLogicalLevel = indentLevelMarkers.length - 1;
                }
            }

            // Use tab characters for standardized indentation
            const newIndent = '\t'.repeat(currentLogicalLevel);
            newLines.push(newIndent + trimmedLine);
        } else {
            // Non-list line, reset context and preserve as-is
            indentLevelMarkers.length = 0;
            newLines.push(line);
        }
    }

    doc.setValue(newLines.join('\n'));
}
```

This implementation successfully handles the logical hierarchy detection and tab-based reformatting described in the feature request.

**Comment by breakid** [comment]
I like this idea. I accidentally ran Markdownlint on my notes once, and it changed (most) of my 4-space indentation to 2-space indentation. Fixing it manually was a pain.

That said, rather than hardcoding a tab character, I think it should use the built-in Obsidian indentation preference for tabs versus spaces.
